src/plugins/template/src/config.cpp

```diff
@@ -85,6 +85,45 @@ Configuration::Configuration(const ov::AnyMap& config, const Configuration& defa
             model_priority = value.as<ov::hint::Priority>();
         } else if (ov::cache_encryption_callbacks == key) {
             encryption_callbacks = value.as<EncryptionCallbacks>();
+        } else if (ov::hint::scheduling_core_type == key) {
+            try {
+                schedulingCoreType = value.as<ov::hint::SchedulingCoreType>();
+            } catch (const std::exception&) {
+                OPENVINO_THROW("Wrong value ", value.as<ov::hint::SchedulingCoreType>(), "for property key ", key);
+            }
+        } else if (key == ov::hint::enable_hyper_threading.name()) {
+            try {
+                enableHyperThreading = value.as<bool>();
+                changedHyperThreading = true;
+            } catch (ov::Exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               value.as<std::string>(),
+                               "for property key ",
+                               ov::hint::enable_hyper_threading.name(),
+                               ". Expected only true/false.");
+            }
+        } else if (key == ov::hint::enable_cpu_pinning.name()) {
+            try {
+                enableCpuPinning = value.as<bool>();
+                changedCpuPinning = true;
+            } catch (ov::Exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               value.as<std::string>(),
+                               "for property key ",
+                               ov::hint::enable_cpu_pinning.name(),
+                               ". Expected only true/false.");
+            }
+        } else if (ov::compilation_num_threads.name() == key) {
+            int val;
+            try {
+                val = value.as<decltype(ov::compilation_num_threads)::value_type>();
```

Comment:
```suggestion
                val = value.as<int>();
```
If property is numerical it should be converted